scripts/common.sh

```diff
@@ -29,10 +34,16 @@ elif [ "$uname" = "FreeBSD" ]; then
     )
 fi
 
+if [ $(uname) = "Darwin" ]; then
+    export LDFLAGS="-Wl,-export_dynamic"
+else
+    export LDFLAGS="-Wl,--export-dynamic"
+fi
```

Comment:
No, I certainly don't want this for all packages, only for guile and lilypond (because it links guile). I'd suggest a new variable (not exported) like `LDFLAGS_EXPORT_DYNAMIC` that is used when needed. You can probably add the flags for `WHOLE_ARCHIVE` here.

(Eventually, this will go away with Guile 2.2 because they don't have srfi modules in shared libraries anymore.)

Reply:
Ok, I see. Defining this as `LDFLAGS_EXPORT_DYNAMIC` seems like a good solution. Will fix.

Reply:
Actually, we should probably not `export LDFLAGS` anywhere. Wouldn't it be better to do add `LDFLAGS="$LDFLAGS_EXPORT_DYNAMIC $LDFLAGS"` as an argument to `configure` those places we want to adjust linker arguments?

Reply:
AFAICS the only place is in `build_lilypond.sh` which should be ok for only building LilyPond itself. Though it's probably an oversight, could be fixed